docs/docs/typescript/generated-types.md

```diff
@@ -0,0 +1,157 @@
+4. types for testing, `currentUser`, etc.
+
+## CurrentUser
+
+If you've setup auth, the type for the current user on both the web and the api side gets automatically "inferred" from the `getCurrentUser` function in `api/src/lib/auth.ts`.
+
+For example, if you specify the return type on `getCurrentUser` as...
+
+```ts title="api/src/lib/auth.ts"
+interface MyCurrentUser {
+  id: string,
+  roles: string[],
+  email: string,
+  projectId: number
+}
+
+const getCurrentUser = ({decoded}): MyCurrentUser => {
+  //..
+}
+```
+
+The types for both `useAuth().currentUser` on the web side and `context.currentUser` on the api side will be the same—the `MyCurrentUser` interface.
+
+## Query and Mutation types
+
+Let's say you have a query in a Cell that looks like this:
+
+```js title="web/src/components/BlogPostCell.tsx"
+export const QUERY = gql`
+  # 👇 Make sure to name your GraphQL operations
+  query FindBlogPostQuery($id: Int!) {
+    blogPost: post(id: $id) {
+      title
+      body
+    }
+  }
+`
+```
+
+Redwood generates types for both the data returned from the query and the query's variables.
+These generated types will use the query's name—in this case, `FindBlogPostQuery`—so you can import them like this:
+
+```ts title="web/src/components/BlogPostCell.tsx"
+import type { FindBlogPostQuery, FindBlogPostQueryVariables } from 'types/graphql'
+```
+
+`FindBlogPostQuery` is the type of the data returned from the query (`{ title: string, body: string }`) and `FindBlogPostQueryVariables` is the type of the query's variables (`{ id: number }`).
+
+The import statement's specifier, `'types/graphql'`, is a [mapped path](https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping). First, TypeScript will look for the types in `web/types/graphql.d.ts`; if they're not there, it'll check `types/graphql.d.ts`. Redwood only automatically generates the former. For the latter, see [sharing types between sides](./introduction.md#sharing-types-between-sides).
+
+But don't worry too much. If you use the generators, they template all of this for you!
+
+## Resolver Types
+
+Generated Services include types for query and mutation resolvers:
+
+```ts title="api/src/services/posts.ts"
+// highlight-next-line
+import type { QueryResolvers, MutationResolvers } from 'types/graphql'
+
+import { db } from 'src/lib/db'
+
+// highlight-next-line
+export const posts: QueryResolvers['posts'] = () => {
+  return db.post.findMany()
+}
+
+// highlight-next-line
+export const post: QueryResolvers['post'] = ({ id }) => {
+  return db.post.findUnique({
+    where: { id },
+  })
+}
+```
+
+These types help you by making sure you're returning an object in the shape of what you've defined in your SDL. Note that these types expect you to return the _complete_ type that you've defined in your SDL. You can just return the result of the Prisma query, and not have to worry about how, for example, a DateTime in Prisma maps to a String in GraphQL.
```

Comment:
I don't quite understand the last two sentences in this paragraph.

Reply:
What I meant to say is - "normally typing these things in graphql is a royal pain in the butt, because you can't just return a DateTime from Prisma (because graphql wants a string for a Date field).... but with Redwood, its all good, and these QueryResolver types already do this mapping for you" 

But I agree doesn't read too well.